### ISSUE TITLE
In Chat mode, undoing an action and then submitting a blank action corrupts the action history

### ISSUE THREAD

**Issue by pi6am** [henk717/KoboldAI-Client#260]
This may be part of a larger issue of action indices getting corrupted, but it's 100% reproducible for me so it may be an easy way to debug that issue. It might instead be chat specific due to the way we decide when to insert "You:".

The issue happens in both classic and new UI mode, but the visuals manifest differently in classic UI, new UI with Legacy mode, and new UI with Messages mode.

To reproduce:
1) Start the server and the client.
2) Enable Chat mode.
3) Use a chat-tailored prompt to begin chatting.
4) Set the Output Length to a moderately low value (e.g. 20-50 tokens)
5) Wait until the AI generates a partial response.
6) Submit an empty action so that the AI continues its response.
7) Use "Undo" to go back one action.
8) Submit an empty action again.
9) Note that the AI does not continue its response. Or if it does generate text it does so in a garbled manner. In the new UI with the Messages chat style, an response from "System" is generated with the text You, and if you continue again then this updates to be a message from You with the text You.

At this point, if you undo the "You" text isn't undone, but instead the previous action is removed. Redoing and generating again will cause previous actions to be replaced. The more you undo and redo, the more corrupted the action state appears to become.

Screenshot of corrupted actions with a You message from You
![image](https://user-images.githubusercontent.com/34464159/212250355-b145e912-2a87-4b50-bc8e-a432cf348891.png)

Screenshot of the context. Notice that the speech attributions "You:" and "Eliza:" in latter part of the context are garbled or missing. This results in the AI responses being incoherent.
![image](https://user-images.githubusercontent.com/34464159/212250566-043c2b37-f6c3-4768-b1af-7f43dd18b75d.png)

**Comment by TheZennou** [comment]
Yup, can confirm this an issue, dealt with it for a long time before jumping ship to Tavern.

**Comment by jojorne** [comment]
Maybe related to https://github.com/henk717/KoboldAI/pull/303.

**Comment by ebolam** [comment]
This should be fixed in PR #333
